cluster_management/src/main/java/com/pinterest/rocksplicator/Spectator.java

```diff
@@ -116,25 +119,34 @@ public static void main(String[] args) throws Exception {
     ));
     CommandLine cmd = processCommandLineArgs(args);
     final String zkConnectString = cmd.getOptionValue(zkServer);
-    final String clusterName = cmd.getOptionValue(cluster);
+    final String[] clusterNamesList = cmd.getOptionValues(clusterNames);
     final String host = cmd.getOptionValue(hostAddress);
     final String port = cmd.getOptionValue(hostPort);
     final String postUrl = cmd.getOptionValue(configPostUrl);
     final String instanceName = host + "_" + port;
 
     LOG.error("Starting spectator with ZK:" + zkConnectString);
-    Spectator spectator= new Spectator(zkConnectString, clusterName, instanceName);
-
     CuratorFramework zkClient = CuratorFrameworkFactory.newClient(zkConnectString, new ExponentialBackoffRetry(1000, 3));
     zkClient.start();
-    InterProcessMutex mutex = new InterProcessMutex(zkClient, getClusterLockPath(clusterName));
-    try (Locker locker = new Locker(mutex)) {
-      spectator.startListener(postUrl);
-      Thread.currentThread().join();
-    } catch (RuntimeException e) {
-      LOG.error("RuntimeException thrown by cluster " + clusterName, e);
-    } catch (Exception e) {
-      LOG.error("Failed to release the mutex for cluster " + clusterName, e);
+
+    int i = 0;
+    while (true) {
+      String currentClusterName = clusterNamesList[i];
+      LOG.error("Trying to get lock for cluster " + currentClusterName);
+      InterProcessMutex mutex = new InterProcessMutex(zkClient, getClusterLockPath(currentClusterName));
+      try (Locker locker = new Locker(mutex, 5, TimeUnit.SECONDS)) {
+        Spectator spectator= new Spectator(zkConnectString, currentClusterName, instanceName);
```

Comment:
Will there be a quick way to tell (from the logs or other means) which host is spectating which cluster? This will help in reducing the debug time when there is an issue with a cluster.

Reply:
you could just look at the logs. I think its the same as you'd have to do right now to find the spectator for rockstore-shared-readonly